### PR TITLE
Fix extconf.rb

### PR DESCRIPTION
**Pull request by soutaro**
See https://github.com/gverger/cbc-wrapper/issues/2#issuecomment-425645135.

* Support for `--with-cbc-*` option for non-standard locations
* Link only `libCbcSolver` to avoid parameter parsing error message

### CODE REVIEW

File: ext/cbc-wrapper/cbc_wrap.c
Comment (gverger) [inline ext/cbc-wrapper/cbc_wrap.c:1854]: This file is automatically generated by `swig`. These 2 lines are defined in `cbc_wrap.i` that `swig` uses as its input. It would probably be good to update the `cbc_wrap.i` file too.
I don't think it is necessary to rerun swig in this case though.

File: ext/cbc-wrapper/extconf.rb
Comment (gverger) [inline ext/cbc-wrapper/extconf.rb:9]: I don't understand how you would use that options `with-cbc-*` ?

**Review by gverger — CHANGES_REQUESTED** [review]
Thank you very much for the time spent in the discovery of this bug. It has been bothering me for a while now!

I will gladly merge this PR once you update the `cbc_wrap.i` file.

This is just a temporary branch though, I think I'll try to make sure installing the gem can use either the system lib or compile it itself.

File: ext/cbc-wrapper/extconf.rb
Comment (soutaro) [reply ext/cbc-wrapper/extconf.rb:9]: If the libraries are not located in the standard path like `/usr/lib` or `/usr/local/lib`, we have to tell `extconf.rb` where the libraries can be found.

Example:

```
$ ruby extconf.rb --with-cbc-dir=/root/Cbc/build
```

**Comment by soutaro** [comment]
@gverger What do you think about using the system library? If you believe it is good, I will revise. If you think building the library during installation is better (maybe on Heroku?), I will close this PR.

**Comment by gverger** [comment]
I think that if I had Cbc installed and I wanted to use this gem, I would expect it to use my installed version instead of spending 10-15 minutes compiling a new one.

I originally recompiled the lib because we were using it on Heroku, and I feel like if it is difficult to set up on Heroku, that would make people not willing to use it.

Eventually, what I would like is to have both possibilities. For now it is fine with me to have this branch for when people want to use their installed lib, so if you fix it, I am happy to merge it.

When I will have some time (I have no idea when), I will probably kind of merge this branch and master to provide different options, but I know that it might be in a while.

File: ext/cbc-wrapper/cbc_wrap.c
Comment (soutaro) [reply ext/cbc-wrapper/cbc_wrap.c:1854]: Updated! I have updated `cbc.i` and confirmed this line is generated by `swig` command (with many extra diffs because my `swig` is 3.0.12.)

**Review by gverger — APPROVED** [review]
Seems good to me! Please merge when you want.

edit: can you merge? if you cannot I'll do it 😄

**Comment by soutaro** [comment]
@gverger I cannot merge. Please click the merge button!

**Pull request merged**
